src/ArticleWrapper.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\fb_instant_articles\ArticleWrapper.
+ */
+
+namespace Drupal\fb_instant_articles;
+
+use Facebook\InstantArticles\Elements\InstantArticle;
+
+/**
+ * Wraps a FB Instant SDK article object with Drupal Base module configs and
+ * hooks for extensibility (without class inheritance, which the FB Instant SDK
+ * prevents by design).
+ *
+ * Class ArticleWrapper
+ * @package Drupal\fb_instant_articles\ArticleWrapper
+ */
+class ArticleWrapper {
+
+  /**
+   * A stateful FB Instant Articles SDK singleton for Drupal integration.
+   *
+   * @see getArticle()
+   *
+   * @var \Facebook\InstantArticles\Elements\InstantArticle
+   */
+  private $instantArticle;
+
+  /**
+   * ArticleWrapper constructor.
+   *
+   * Instantiates the InstantArticle object, and adds Drupal Base module configs
+   * where appropriate.
+   *
+   * Note modules making use of this wrapper must set the required Canonical URL
+   * with:
+   * @code
+   * ArticleWrapper::create()->getArticle()->withCanonicalUrl($url);
+   * @endcode
+   */
+  private function __construct() {
+    $this->instantArticle = InstantArticle::create()
+      ->addMetaProperty('op:generator:application', 'drupal/fb_instant_articles')
+      ->addMetaProperty('op:generator:application:version', self::getApplicationVersion())
+      ->withStyle(variable_get('fb_instant_articles_style', 'default'));
+
+  }
+
+  /**
+   * Creates a Drupal wrapper for an InstantArticle object.
+   *
+   * @return \Drupal\fb_instant_articles\ArticleWrapper
+   */
+  public static function create() {
+    return new ArticleWrapper();
+  }
+
+  /**
+   * Gets the wrapped InstantArticle object.
+   *
+   * Also invokes a hook to allow other modules to alter the InstantArticle
+   * object before render or any other operation.
+   *
+   * @see hook_fb_instant_articles_instant_article_alter()
+   *
+   * @return \Facebook\InstantArticles\Elements\InstantArticle
+   */
+  public function getArticle() {
+    drupal_alter('fb_instant_articles_article', $this->instantArticle);
```

Comment:
@m4olivei you're right, I actually meant to add a `$context` param to the hook as well. This will allow modules to add arbitrary contextual information to the `ArticleWrapper` for altering the InstantArticle object. Hang on and I'll add a commit for this.